samplesheets/tests/test_views_taskflow.py

```diff
@@ -1830,6 +1839,336 @@ def test_accept_lock_failure(self):
+    """Tests for IrodsRequestAcceptBatchView"""
+
+    def test_render(self):
+        """Test rendering IrodsRequestAcceptBatchView"""
+        self.assert_irods_obj(self.path)
+        self.assert_irods_obj(self.path_md5)
+
+        with self.login(self.user_contrib):
+            self.client.post(
+                reverse(
+                    'samplesheets:irods_request_create',
+                    kwargs={'project': self.project.sodar_uuid},
+                ),
+                self.post_data,
+            )
+            self.assertEqual(IrodsDataRequest.objects.count(), 1)
+            self.client.post(
+                reverse(
+                    'samplesheets:irods_request_create',
+                    kwargs={'project': self.project.sodar_uuid},
+                ),
+                self.post_data2,
+            )
+            self.assertEqual(IrodsDataRequest.objects.count(), 2)
+
+        with self.login(self.user):
+            response = self.client.post(
+                reverse(
+                    'samplesheets:irods_request_accept_batch',
+                    kwargs={'project': self.project.sodar_uuid},
+                ),
+                {
+                    'irodsdatarequests': ','.join(
+                        [
+                            str(irods_request.sodar_uuid)
+                            for irods_request in IrodsDataRequest.objects.all()
+                        ]
+                    ),
+                },
+            )
+            self.assertEqual(response.status_code, 200)
+
+        self.assertEqual(
+            response.context['affected_object_paths'],
+            sorted(
+                set(
+                    [
+                        irods_request.path
+                        for irods_request in IrodsDataRequest.objects.all()
+                    ]
+                )
+            ),
+        )
+        self.assertEqual(len(response.context['irods_request_data']), 2)
+        self.assertEqual(
+            response.context['irods_request_data'][0],
+            IrodsDataRequest.objects.first(),
+        )
+        self.assertEqual(
+            response.context['irods_request_data'][1],
+            IrodsDataRequest.objects.last(),
+        )
+
+    def test_render_coll(self):
+        """Test rendering IrodsRequestAcceptBatchView for a collection"""
+        coll_path = os.path.join(self.assay_path, 'request_coll')
+        self.irods.collections.create(coll_path)
+        self.assertEqual(self.irods.collections.exists(coll_path), True)
+        self.post_data['path'] = coll_path
+
+        with self.login(self.user_contrib):
+            self.client.post(
+                reverse(
+                    'samplesheets:irods_request_create',
+                    kwargs={'project': self.project.sodar_uuid},
+                ),
+                self.post_data,
+            )
+            self.assertEqual(IrodsDataRequest.objects.count(), 1)
```

Comment:
Remember, we can unindent lines after `with self.login(...)` unless we need to do other stuff which requires that login. Or if we want to assert redirecting.